src/vm/readytoruninfo.cpp

```diff
@@ -788,6 +793,8 @@ PCODE ReadyToRunInfo::GetEntryPoint(MethodDesc * pMD, PrepareCodeConfig* pConfig
 #endif
     }
 
+done:
```

Comment:
I still think it would be better to check here.

Reply:
I just saw your comment about EX_TRY.  What about doing this:

```
    if (ETW_EVENT_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_Context, R2RGetEntryPoint))
    {
        EX_TRY
        {
                SString tNamespace, tMethodName, tMethodSignature;
                pMethodDesc->GetMethodInfo(tNamespace, tMethodName, tMethodSignature);

                FireEtwR2RGetEntryPoint(
                    (UINT64)pMethodDesc,
                    (PCWSTR)tNamespace.GetUnicode(),
                    (PCWSTR)tMethodName.GetUnicode(),
                    (PCWSTR)tMethodSignature.GetUnicode(),
                    pEntryPoint,
                    GetClrInstanceId());

        } EX_CATCH{ } EX_END_CATCH(SwallowAllExceptions);
    }
```

I can still put the check inside of GetEntryPoint if you want, but this should reduce the cost pretty well I think.

Reply:
On second thought, you're probably right given how often this gets called.